### PR TITLE
[openstack] Ensure openstack passwords and secrets are obfuscated.

### PR DESCRIPTION
**Pull request by lyarwood**
- Add a postproc method to ceilometer, cinder, glance, hoirzon and swift.
- Add missing keys to the remaining plugins.
- Modify the regular expression used by all plugins to ignore commented
  out keys.

Resolves: #574

Signed-off-by: Lee Yarwood <lyarwood@redhat.com>

### CODE REVIEW

File: sos/plugins/openstack_cinder.py [hunk 1]
```diff
@@ -42,6 +42,24 @@ def setup(self):
...
             self.add_copy_spec(["/var/log/cinder/"])
 
+    def postproc(self):
+        protect_keys = [
+            "admin_password", "backup_tsm_password", "chap_password",
+            "nas_password", "cisco_fc_fabric_password", "coraid_password",
+            "eqlx_chap_password", "fc_fabric_password",
+            "hitachi_auth_password", "hitachi_horcm_password",
+            "hp3par_password", "hplefthand_password", "memcache_secret_key",
+            "nas_password", "netapp_password", "netapp_sa_password",
```
Comment (pmoravec) [inline sos/plugins/openstack_cinder.py]: nas_password is here twice (cf. line 48).

File: sos/plugins/openstack_neutron.py [hunk 2]
```diff
@@ -61,10 +60,10 @@ def postproc(self):
             "ssl_key_password", "nsx_password", "vcenter_password",
             "edge_appliance_password", "tenant_admin_password", "apic_password"
         ]
-        regexp = r"((?m)^\s*#*(%s)\s*=\s*)(.*)" % "|".join(protect_keys)
+        regexp = r"((?m)^\s*(%s)\s*=\s*)(.*)" % "|".join(protect_keys)
```
Comment (pmoravec) [inline sos/plugins/openstack_neutron.py]: When doing a change here, could you also please remove duplicate `qpid_password` in `protected_key` (1st and 3rd line of the list)?

File: sos/plugins/openstack_nova.py [hunk 3]
```diff
@@ -67,12 +67,12 @@ def postproc(self):
             "qpid_password", "powervm_mgr_passwd", "virtual_power_host_pass",
             "xenapi_connection_password", "password", "host_password",
             "vnc_password", "connection", "sql_connection", "admin_password"
+            "connection_password", "memcache_secret_key", "s3_secret_key"
```
Comment (pmoravec) [inline sos/plugins/openstack_nova.py]: Shouldn't previous line (69) and this one have coma at the end?

File: sos/plugins/openstack_trove.py [hunk 4]
```diff
@@ -34,21 +35,13 @@ def setup(self):
...
-        ]
-
-        conf_list = [
-            '/etc/trove/trove.conf',
-            '/etc/trove/trove-conductor.conf',
-            '/etc/trove/trove-guestmanager.conf',
-            '/etc/trove/trove-taskmanager.conf'
+            "default_password_length", "notifier_queue_password",
+            "rabbit_password", "replication_password", "connection",
+            "admin_password", "dns_passkey"
```
Comment (pmoravec) [inline sos/plugins/openstack_trove.py]: You are removing from protect_keys these params/items:

nova_proxy_admin_pass
qpid_password
sql_connection

Is the removal safe or can the parameters have values/passwords to be obfuscated?

Comment (lyarwood) [inline sos/plugins/openstack_trove.py]: This is safe, AFAICT nova_proxy_admin_pass isn't required, qpid_password isn't supported and sql_connection is the legacy version of connection.

File: sos/plugins/openstack_nova.py (hunk 3 above)
Comment (lyarwood) [inline sos/plugins/openstack_nova.py]: Yup, updating now.

File: sos/plugins/openstack_neutron.py (hunk 2 above)
Comment (lyarwood) [inline sos/plugins/openstack_neutron.py]: ACK.

File: sos/plugins/openstack_cinder.py (hunk 1 above)
Comment (lyarwood) [inline sos/plugins/openstack_cinder.py]: ACK.

**Comment by pmoravec** [comment]
Sounds fine from my side. :+1:

**Pull request merged**